### PR TITLE
Adopt MkDocs and use Read the Docs theme

### LINKED ISSUE #102: Choose another documentation theme
### Requested theme
- https://github.com/readthedocs/sphinx_rtd_theme

### Alternative themes to consider
- https://github.com/just-the-docs/just-the-docs
- https://github.com/mkdocs/mkdocs

### Recommendations for contributors
- Avoid breaking existing pages and current links such as https://sandboxie-plus.github.io/sandboxie-docs/Content/ResourceAccess.html
- If the suggested themes are not of your liking, feel free to suggest a new one

This issue is not in my plans, but if someone wanted to attach a pull request, that would be cool.

### PR DESCRIPTION
**Pull request by QZLin**
https://qzlin.github.io/sandboxie-docs

### CODE REVIEW

**Comment by isaak654** [comment]
Unless you click the plus buttons, it is not possible to directly access "Sandboxie Control" and "SBIE Messages" with Firefox 124.0.2 and MS Edge 123.0.2420.81:

https://github.com/sandboxie-plus/sandboxie-docs/assets/12372772/b56d7b46-969a-40f5-af89-2f6eb5810a04

**Comment by QZLin** [comment]
That seems the expected behaviour of the mkdocs default readthedoc theme. I will try to change it.

**Comment by isaak654** [comment]
1. If you go into the "Usage Tips" page, the minus button is still displayed even when there is nothing below to collapse
2. The Sandboxie logo feels empty without a `sandboxie-docs` text next to it.
3. Search highlighting doesn't seem to be supported in the theme, so this can be skipped and postponed later

**Comment by isaak654** [comment]
1. If you go into the "Usage Tips" page, the minus button is still displayed even when there is nothing below to collapse
2. The Sandboxie logo feels empty without a `sandboxie-docs` text next to it
3. Search highlighting doesn't seem to be supported in the theme, so this can be skipped and postponed later

File: docs/_layouts/default.html
Comment (isaak654) [inline docs/_layouts/default.html:1]: One of the old paths that needs cleaning up.

File: docs/assets/css/style.scss
Comment (isaak654) [inline docs/assets/css/style.scss:1]: One of the old paths that needs cleaning up.

File: docs/PlusContent/WFPSupport.md
Comment (isaak654) [inline docs/PlusContent/WFPSupport.md:1]: Nothing against moving Content and Media to docs, but I have the impression that all other pending pull requests will end up with broken references, unless this pull request is merged for last. Can you confirm this?

File: custom_theme/css/theme_extra.css [hunk 1]
```diff
@@ -236,4 +236,7 @@ code {
 /*  */
 .logo {
     width: 100px !important;
+}
+.logo-label {
+    color: #f8f8f8;
```
Comment (isaak654) [inline custom_theme/css/theme_extra.css]: Please add:

```
label {
    font-size:100%;
}
```

```
.wy-side-nav-search .wy-dropdown>a,
.wy-side-nav-search>a {
    font-weight:normal;
}
```

```
.wy-nav-top a {
    font-weight:normal;
}
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+    # paths:
+# - docs/**
+# - custom_theme
+# - README.md
+# - requirements.txt
```
Comment (isaak654) [inline .github/workflows/publish.yml]: Can they be removed or do they have a later purpose?

**Review by isaak654 — CHANGES_REQUESTED** [review]

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  name: readthedocs
+  custom_dir: custom_theme
+  logo: https://github.com/sandboxie-plus/Sandboxie/blob/master/SandboxiePlus/SandMan/Resources/SandMan.png?raw=true
+  highlightjs: true
+  sticky_navigation: true
+  collapse_navigation: true
+watch:
+- custom_theme
+copyright: 'sandboxie-docs is maintained by <a href="https://github.com/sandboxie-plus">sandboxie-plus</a>. If you want to contribute: <a href="https://github.com/sandboxie-plus/sandboxie-docs/edit/main/README.md">Improve this page!</a>'
+nav:
```
Comment (isaak654) [inline mkdocs.yml:19]: Is the entire .md file structure automatically updated when a new file is found under the docs folder?

Comment (QZLin) [reply mkdocs.yml:19]: Yes, page is automatically generated for new `.md` file in docs. Nav requires manual update.
The `watch` parameter only affects the live reload of the development, and the `mkdocs.yml` file, folder `docs` will always be watched.

File: docs/PlusContent/WFPSupport.md
Comment (QZLin) [reply docs/PlusContent/WFPSupport.md:1]: Args of lychee-action is modified for changed doc structure
https://github.com/QZLin/sandboxie-docs/actions/runs/8642656793

**Review by QZLin — COMMENTED** [review]
Is the label of logo no longer required?

**Comment by isaak654** [comment]
I think it would be better if the text was part of the logo itself.

In this regard, I wonder if it is possible to reproduce a yellow font similar to this one: 
[105014988-575cb680-5a41-11eb-90d5-d942d28184b2.png](https://user-images.githubusercontent.com/3890945/105014988-575cb680-5a41-11eb-90d5-d942d28184b2.png)

**Comment by QZLin** [comment]
a merged logo
sandmerge

**Comment by isaak654** [comment]
I'm thinking of something more recognizable like this:

342063615-87df6d8d-4a1c-4ddd-b248-5c20e42b84d7_v3

where "docs" is Segoe UI Bold 24 pt

### Old result (.logo width is set to 200px):

logo preview


### Graphic defects:

- I cannot make the black border (the outline) of the word "docs" similar to that in the word "Sandboxie"
- The grey mark near to the S of Sandboxie (also present in your merged logo):
S letter of Sandboxie logo

---
New attempts with "docs" set as centered and with font size 30 pt:

"docs" set as centered and with font size 30 pt "docs" set as centered and with font size 30 pt (yellow variant)

---
New attempts (text only, without icon):

Sandboxie text only (logo) Sandboxie text only (logo v2)

**Comment by QZLin** [comment]
Cleaned up the leftover pixels and add shadow to "docs" text
sandmerge
Preview with background color:
preview

**Comment by isaak654** [comment]
Could you help me to fix the following points before merging?

- [ ] 1) Apply [this](https://private-user-images.githubusercontent.com/12372772/342458489-e854dcc6-361f-4d50-bdab-b16500425f4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIzNjg4MzAsIm5iZiI6MTcyMjM2ODUzMCwicGF0aCI6Ii8xMjM3Mjc3Mi8zNDI0NTg0ODktZTg1NGRjYzYtMzYxZi00ZDUwLWJkYWItYjE2NTAwNDI1ZjRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMwVDE5NDIxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlZDg3ZDU3ODJhNmRlNjZjNDZhMmYzMGJjN2ZmODQ2YTEzZWNjYjY5OWVkMmNkODgxNThkZjJjOTZmZjdiZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lFhm5WDd8iPPTb2qM0UZQWjH73VDFS7KW72WDrkU_ro) logo
- [ ] 2) Apply the correct width for .logo CSS class
- [ ] 3) Fix the conflicting files

**Comment by isaak654** [comment]
I have merged the previous PRs and decided which logo to adopt.

Could you help me to fix the following points before merging?

- [x] 1) Apply [this](https://private-user-images.githubusercontent.com/12372772/342458489-e854dcc6-361f-4d50-bdab-b16500425f4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0MTQ3MzYsIm5iZiI6MTcyMjQxNDQzNiwicGF0aCI6Ii8xMjM3Mjc3Mi8zNDI0NTg0ODktZTg1NGRjYzYtMzYxZi00ZDUwLWJkYWItYjE2NTAwNDI1ZjRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMxVDA4MjcxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiMmI2Y2MyMWRiZjFlN2UwMGU2NzI2N2VlZTQwOTY2MGQzYzQxOGFhMWVmNWY4N2NhMTQwN2U0YTVmOWQ4M2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.mUtUk5LHc3iw6Gkn5_zgI5ZEJSZonilnC6vx_wc8btQ) logo
- [x] 2) Remove width for .logo CSS class
- [x] 3) Fix the conflicting files

**Comment by QZLin** [comment]
> https://github.com/sandboxie-plus/sandboxie-docs/pull/167#issuecomment-2259102580

The logo link seems broken (Got 404 not found)

**Comment by isaak654** [comment]
> The logo link seems broken (Got 404 not found)

Not anymore, I've just updated the link.

**Comment by QZLin** [comment]
https://github.com/QZLin/sandboxie-docs/commit/85fd13c67336b62588401c4dbad68ca5d4ed92b1
I'm not sure if splitting the readme into repo readme and docs readme is a good idea.
But the links of readme.md and docs/readme.md have to be different because of the mkdocs project structure.Then we will have to always do the sync jobs for two similar files.
```
// readme.md
[Getting Started with Sandboxie](docs/Content/GettingStarted.md)

// docs/readme.md
[Getting Started with Sandboxie](Content/GettingStarted.md)
```

**Comment by isaak654** [comment]
> I'm not sure if splitting the readme into repo readme and docs readme is a good idea.

I think renaming docs/readme.md to docs/index.md could help to prevent confusion, while in the root README:

```
# Sandboxie documentation

README is located at [docs/index.md](docs/index.md).
```

A second alternative  would be the content of: 
https://github.com/okrc/sandboxie-docs/blob/cd3e116b3fb817386eee5edc26fb99bb5d3dfe83/docs/index.md

but to make a decision we should have a list of pros and cons.

**Comment by isaak654** [comment]
The search box under the logo suddenly disappeared.

**Comment by QZLin** [comment]
> The search box under the logo suddenly disappeared.

Fixed

**Review by isaak654 — APPROVED** [review]

**Pull request merged**